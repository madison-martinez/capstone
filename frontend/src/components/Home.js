import styled from 'styled-components';
import Image from 'next/image';
import Link from 'next/link';
import Router from 'next/router'
import { useUser } from './User';
import { useEffect } from 'react';


const GalleryStyles = styled.div`
    display: grid;
    grid-template-columns: repeat(3, 1fr);
    grid-gap: 1rem;
    .text-center {
        text-align: center;
    }
    .buttons-section {
        display: grid;
        grid-template-columns: repeat(2, 1fr);
        grid-gap: 2rem;
        font-size: 2rem;
        a {
            &:hover {
                background-color: ${props => props.theme.blue};
                color: ${props => props.theme.offWhite};
            }
        }
    }
    @media (max-width: 700px) {
        display: grid;
        grid-template-columns: 1fr;
    }
`;


function Home() {
    const currentUser = useUser();
    console.log(currentUser);

    useEffect(() => {
    
    }, [])

    return (
        <GalleryStyles>
            <Image
                src="/assets/photo-1515150144380-bca9f1650ed9.jpeg"
                alt="farmer watering plant"
                layout="responsive"
                width={600}
                height={900}
            />
            <div className="text-center">
                <h2>What We Do</h2>
                <p>
                    We are trying to be a link in a somewhat broken supply chain.
                    We are looking to help the small providers get their products into the locals hands and bellies.
                    </p>
                <h2>Why We Do It</h2>
                <p>
                    While the food supply chain has been stressed for awhile, the COVID pandemic has really brought
                    to light the connection between farmers and consumers, especially small businesses.
                    Instead of having farmers dump milk or plow fields full of food, we hope to maybe find places for that food to go, while keeping the farmers fed too.
                    </p>
                <h2>Where You Fit In</h2>
                <p>
                    The goal of this website is to allow local producers to list items they have for sale to local restaurants who may be looking to get a larger margin
                    on dishes during this difficult time. So, if you would like to see what is available to you, please sign up or login and we look forward to seeing
                    what you create.
                    </p>
                <div className="buttons-section">
                    {currentUser && (
                        <Link href="/login">
                            <a>Logout</a>
                        </Link>
                    )}
                    {!currentUser && (
                        <>
                            <Link href="/signup">
                                <a>Signup</a>
                            </Link>
                            <Link href="/login">
                                <a>Login</a>
                            </Link>
                        </>
                    )}
                </div>
            </div>
            <Image
                src="/assets/photo-1503453776591-b4548af666a2.jpeg"
                alt="kitchen worker salting food"
                layout="responsive"
                width={600}
                height={900}
            />
        </GalleryStyles>
    )
};

export default Home;
