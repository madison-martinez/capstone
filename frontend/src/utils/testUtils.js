import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { reducers } from '../reducers';

function render(
  ui,
  {
    initialState,
    store = createStore(reducers, initialState ),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
}

//mock middleware functions
const thunk = ({ dispatch, getState }) => next => action => {
    if (typeof action === 'function') {
      return action(dispatch, getState)
    }
    return next(action)
};

const fakeItem = () => ({
    id: '1',
    price: 20.00,
    title: 'Roses',
    description: 'Roses are Red',
});

export {
    fakeItem,
    thunk
};

export * from '@testing-library/react';
export { render };