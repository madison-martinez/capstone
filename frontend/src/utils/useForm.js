import { useState } from 'react';



export default function useForm(initialState = {}) {
    const [values, setValues] = useState(initialState);

    function handleChange(e) {
        let { value, name, type } = e.target;
        if (type === 'number') {
            value = parseInt(value);
        }
        if (type === 'file') {
            [value] = e.target.files;
        }
        setValues({
            ...values,
            [name]: value
        });
    }

    function clearForm() {
        const emptyForm = Object.fromEntries(
            Object.entries(values).map(([key]) => [key, ''])
        );
        setValues(emptyForm);
    }

    function setInitialForm() {
        setValues(initialState);
    }

    return {
        values, 
        handleChange, 
        clearForm, 
        setInitialForm
    }
}