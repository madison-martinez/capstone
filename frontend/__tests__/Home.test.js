import { render, screen, fakeUser } from '../src/utils/testUtils';
import Home from '../src/components/Home';

const user = fakeUser();
describe('<Home />', () => {
    it('should render home page with no user', () => {
        const { container } = render(
            <Home/>, { initialState: { authentication: {user: {}}}})
        expect(container).toMatchSnapshot();
    });
    it('shows correct buttons if logged in', () => {
        render(
        <Home/>, { initialState: { authentication: { user: {user}}}})
        expect(screen.getByText('Logout')).toBeInTheDocument();
    });
});
