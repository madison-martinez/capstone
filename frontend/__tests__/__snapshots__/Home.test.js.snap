// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render home page with no user 1`] = `
<div>
  <div
    class="Home__GalleryStyles-sc-1ig96v5-0 jmUsjy"
  >
    <div
      style="display: block; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
    >
      <div
        style="display: block; box-sizing: border-box; padding-top: 150%;"
      />
      <img
        alt="farmer watering plant"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
      />
    </div>
    <div
      class="text-center"
    >
      <h2>
        What We Do
      </h2>
      <p>
        We are trying to be a link in a somewhat broken supply chain. We are looking to help the small providers get their products into the locals hands and bellies.
      </p>
      <h2>
        Why We Do It
      </h2>
      <p>
        While the food supply chain has been stressed for awhile, the COVID pandemic has really brought to light the connection between farmers and consumers, especially small businesses. Instead of having farmers dump milk or plow fields full of food, we hope to maybe find places for that food to go, while keeping the farmers fed too.
      </p>
      <h2>
        Where You Fit In
      </h2>
      <p>
        The goal of this website is to allow local producers to list items they have for sale to local restaurants who may be looking to get a larger margin on dishes during this difficult time. So, if you would like to see what is available to you, please sign up or login and we look forward to seeing what you create.
      </p>
      <div
        class="buttons-section"
      >
        <a
          href="/login"
        >
          Logout
        </a>
      </div>
    </div>
    <div
      style="display: block; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
    >
      <div
        style="display: block; box-sizing: border-box; padding-top: 150%;"
      />
      <img
        alt="kitchen worker salting food"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
      />
    </div>
  </div>
</div>
`;
